db/dcrsqlite/sync.go

```diff
@@ -405,3 +409,54 @@ func (db *wiredDB) getBlock(ind int64) (*dcrutil.Block, *chainhash.Hash, error)
 
 	return block, blockhash, nil
 }
+
+// ImportSideChains imports all side chains. Similar to pgblockchain.MissingSideChainBlocks
+// plus the rest from main.go
+func (db *wiredDB) ImportSideChains(collector *blockdata.Collector) error {
+	tips, err := rpcutils.SideChains(db.client)
+	if err != nil {
+		return err
+	}
+	hashlist := make([]*chainhash.Hash, 0)
```

Comment:
When not allocating the slice, we generally do just `var hashlist []*chainhash.Hash`.

Reply:
However, we know the length of `tips`, so we should actually do `hashlist := make([]*chainhash.Hash, 0, len(tips))` to have the right amount of space allocated.